x/gov/endblocker_test.go

```diff
@@ -309,7 +310,7 @@ func TestEndBlockerProposalHandlerFailed(t *testing.T) {
 
 	// Create a proposal where the handler will pass for the test proposal
 	// because the value of contextKeyBadProposal is true.
-	ctx = ctx.WithValue(contextKeyBadProposal, true)
+	ctx = ctx.WithContext(context.WithValue(ctx.Context(), contextKeyBadProposal, true))
```

Comment:
Couldn't we make a simple wrapper for this operation so that we don't break the API here?

Reply:
I could...  I wanted to deprecate this usage and enforce a break to see if it is widely used. I can make this non-breaking API wise.

Will update

Reply:
I'd love to avoid breaking API here.